src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.Connections.cs

```diff
@@ -40,18 +40,25 @@ internal static class Connections
 
                         try
                         {
-                            return await funcAsync().ConfigureAwait(false);
+                            return await client.RequestServiceAsync(descriptor, cancellationToken).ConfigureAwait(false);
                         }
-                        catch (TException)
+                        catch (OperationCanceledException e) when (e.CancellationToken != cancellationToken)
```

Comment:
> e.CancellationToken != cancellationToken

❕ There is no expectation that a cancellation exception will _ever_ have particular token (e.g. cases where a linked token is used). The valid check here is:

```suggestion
                        catch (OperationCanceledException e) when (!cancellationToken.IsCancellationRequested)
```

Reply:
Good catch.

Reply:
this doesn't seem to have been resolved.